pm/sigverifier_test.go

```diff
@@ -0,0 +1,76 @@
+package pm
+
+import (
+	"fmt"
+	"testing"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+func TestVerify(t *testing.T) {
+	msg := []byte("foo")
+	personalMsg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", 32, msg)
+	personalHash := crypto.Keccak256([]byte(personalMsg))
+
+	senderPrivKey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sender := crypto.PubkeyToAddress(senderPrivKey.PublicKey)
+
+	senderSig, err := crypto.Sign(personalHash, senderPrivKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	signerPrivKey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	signer := crypto.PubkeyToAddress(signerPrivKey.PublicKey)
+
+	signerSig, err := crypto.Sign(personalHash, signerPrivKey)
+
+	b := &stubBroker{
+		usedTickets:     make(map[ethcommon.Hash]bool),
```

Comment:
Why do we need to instantiate `usedTickets` for this test?

Reply:
I've replaced declarations of `stubBroker` struct literals with a call to the constructor `newStubBroker` which creates a `stubBroker` with all fields initialized and ready for use.

My thought was to try to avoid any accidental usage of `stubBroker` that leads to accessing a nil map.